docs/config-file/v2.rst

```diff
@@ -0,0 +1,365 @@
+~~~~~~~~~~
+
+Git submodules configuration.
+
+.. note::
+   
+   You can't include and exclude submodules at the same time.
+
+.. code-block:: yaml
+   
+   submodules:
+     include: all
+     recursive: true
+
+submodules.include
+``````````````````
+
+List of submodules to be included.
+
+:Type: ``list``
+:Default: ``[]``
+
+.. note::
+   
+   You can use the ``all`` keyword to include all submodules.
+
+   .. code-block:: yaml
+
+      submodules:
+        include: all
+
+submodules.exclude
+``````````````````
+
+List of submodules to be excluded.
+
+:Type: ``list``
+:Default: ``[]``
+
+.. note::
+   
+   You can use the ``all`` keyword to exclude all submodules.
+   This is the same as ``include: []``.
+
+   .. code-block:: yaml
+
+      submodules:
+        exclude: all
+
+submodules.recursive
+````````````````````
+
+Do a recursive clone of the submodules.
+
+:Type: ``bool``
+:Default: ``false``
+
+Schema
+------
+
+You can see the complete schema
+`here <https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/rtd_tests/fixtures/spec/v2/schema.yml>`_.
+
+Migrating from v1
+-----------------
+
+- First, make sure to add the :ref:`version setting <config-file/v2:version>`.
+
+Changes
+~~~~~~~
+
+- The default value of the :ref:`config-file/v2:formats` setting has change to ``[]``
+  and it doesn't include the values from the web interface.
+- The top setting ``requirements_file`` was moved to ``python.requirements``.
+  And it accepts the ``''`` (empty string) value to completely ignore finding a requirements file.
+  See :ref:`config-file/v2:python.requirements`.
+- The setting `conda.file` was renamed to ``conda.environment``.
+  See :ref:`config-file/v2:conda.environment`.
+- The ``build.image`` setting now only has two options: ``latest`` (default) and ``stable``.
+  See :ref:`config-file/v2:build.image`.
+- The settings ``python.setup_py_install`` and ``python.pip_install`` were replaced by ``python.install``.
+  The previous setting is the same as ``python.install: setup.py`` or ``python.install: pip``.
+  See :ref:`config-file/v2:python.install`.
+- The setting ``python.use_system_site_packages`` was renamed to ``python.system_packages``.
+  See :ref:`config-file/v2:python.system_packages`.
+
+.. warning::
+
+   Some values from the web interface are no longer respected,
+   please see :ref:`config-file/v2:Migrating from the web interface` if you have settings there.
+
+New
+~~~
+
+- :ref:`config-file/v2:sphinx`
+- :ref:`config-file/v2:mkdocs`
+- :ref:`config-file/v2:submodules`
+
+Migrating from the web interface
+--------------------------------
+
+This should be pretty straightforward,
+just go to the admin panel of your project,
+advanced settings and find their respective setting :ref:`here <config-file/v2:Supported settings>`.
+
+.. note::
+
+   Not all settings in the web interface are per version, but are per project.
+   Those settings aren't supported via the configuration file (like ``Default branch``).
```

Comment:
I suppose we should have a specific list with all the settings that can't be managed via the YAML file.

Reply:
Need to check the admin panel p: